docs/ldml/tr35.md

```diff
@@ -2861,38 +2861,52 @@ Element content whose display may be affected in this way should include an expl
+| `propName`     | <pre>= s \[A\-Za\-z0\-9\] \[A\-Za\-z0\-9\_\\x20\]\* s</pre>                | General\_Category,<br/>General Category  |
+| `value1`       | <pre>= \[^\\\}\]<br/>\| '\\' quoted</pre>                         | Lm,<br/>\\n,<br/>\\\}                    |
+| `value2`       | <pre>= \[^:\]<br/>\| '\\' quoted</pre>                           | Lm,<br/>\\n,<br/>\\:                    |
+| `element`      | <pre>= char \| string \| var </pre> | %, b, \{hello\}, \{\}, \\x\{61 62\} |
+| `string`      | <pre>= '\{' \(s charInString\)\* s '\}' </pre> | \{hello\} |
+| `char`         | <pre>= \[^ \\^ \\& \\\- \\\[ \\\] \\\\ \\\{ \\$ \[:Pat_WS:\]\]<br/>\| '\\' quoted</pre> | a, b, c, \\n, \\\{, \\$ |
+| `charInString` | <pre>= \[^ \\\\ \\\} \[:Pat_WS:\]\]<br/>\| '\\' quoted</pre> | a, b, c, \\n, \{, $ <br/> *note: only quoted items that resolve to one code point are allowed* | <!-- TODO: Multi escape semantics? If flattening semantics, this note is unnecessary -->
+| `quoted`       | <pre>= 'u' \(hex\{4\} \| bracketedHex\)<br/>\| 'x' \(hex\{2\} \| bracketedHex\)<br/>\| 'U00' \('0' hex\{5\} \| '10' hex\{4\}\)<br/>\| 'N\{' charName '\}'<br/>\| \[\[\\u0000\-\\U00010FFFF\]\-\[uxUN\]\]</pre> | n, U0000FFFE, \{, $, \] <br/> _note: lengths are exact_ |
+| `charName`     | <pre>= s \[A\-Za\-z0\-9\] \[\-A\-Za\-z0\-9\_\\x20\]\* s</pre>                | TIBETAN LETTER \-A                       |
+| `bracketedHex` | <pre>= '\{' s hexCodePoint \(s hexCodePoint\)\* s '\}'</pre>        | \{61 2019 62\}  <br/> *note: the whitespace between hexCodePoints is required here*                         | <!-- TODO: Multi escape semantics? If so, add note explaining here or somewhere else -->
+| `hexCodePoint` | <pre>= hex\{1,5\} \| '10' hex\{4\}</pre>                           |                                         |
+| `hex`          | <pre>= \[0\-9A\-Fa\-f\]</pre>                                       |                                         |
+| `var`          | <pre>= '$' \[:ID_Start:\] \[:ID_Continue:\]\*</pre>                 | $a, $elt5  (optional support)            | <!-- TODO: XID* or ID*? -->
+| `s`            | <pre>= \[:Pattern_White_Space:\]\*</pre>                          | optional whitespace                     |
+
+Some constraints on UnicodeSet syntax are not captured by this EBNF.
+Notably:
+1. Property names and values are restricted to those supported by the implementation, and have additional constraints imposed by [[UAX44](https://www.unicode.org/reports/tr41/#UAX44)].
+2. Quoted characters in strings must resolve to exactly one character. That is, **\[\{\x{61\}\}\]** is valid and equivalent to **\[\{a\}\]**, but **\[\{\\x\{61 62\}\}\]** is invalid and in particular not equivalent to **\[\\x\{61 62\}\]**, which is equivalent to **\[\{ab\}\]**. <!-- TODO: Multi escape semantics? -->
```

Comment:
Ok, I see where you are coming from earlier.

if ICU permits [\u{23 45}] but does not permit [{\u{23 45}}] then that is probably a problem with ICU's implementation. That is, where supported, \u{23 45} should always be equivalent to \u{23}\u{45}

NOTE: you can't use the online jsp tools for this, because they do some extra unescaping beyond what UnicodeSet supports.

Reply:
Also, you're missing the following from my PR.
> If […] starts with [:, then it begins a prop, and must terminate with :]. Thus [:di:] is a valid property expression, [di:] is a 3 code-point set, and [:di] raises an error.

(You might get to this later on.)

Reply:
I do get to that later on, but I think this is an important enough addition to add it to the notes directly underneath the EBNF. Added it here.

There is some slight duplication between the constraint section underneath the EBNF, and the constraint section underneath the table of syntax characters, however I think the following way to think about that split makes sense:
* The section underneath the EBNF talks about _global_ constraints about the syntax and some semantics
* The section underneath the table of syntax chars talks about _context-sensitive_ constraints on syntax characters

If you disagree, we can try to combine the two constraint sections somehow.